swap/src/bin/swap.rs

```diff
@@ -222,8 +170,135 @@ async fn main() -> Result<()> {
+                    monero_wallet,
+                    config,
+                    db,
+                )
+                .await?;
+            } else if let Ok(bob_state) = BobState::try_from(db_swap) {
+                let (bitcoin_wallet, monero_wallet) = setup_wallets(
+                    bitcoind_url,
+                    bitcoin_wallet_name.as_str(),
+                    monero_wallet_rpc_url,
+                    config,
+                )
+                .await?;
+                bob_swap(swap_id, bob_state, bitcoin_wallet, monero_wallet, db).await?;
+            } else {
+                anyhow::bail!("Unable to construct swap state for swap with id {}")
+            }
+        }
     }
 
     Ok(())
 }
+
+async fn setup_wallets(
+    bitcoind_url: url::Url,
+    bitcoin_wallet_name: &str,
+    monero_wallet_rpc_url: url::Url,
+    config: Config,
+) -> Result<(Arc<bitcoin::Wallet>, Arc<monero::Wallet>)> {
+    let bitcoin_wallet =
+        bitcoin::Wallet::new(bitcoin_wallet_name, bitcoind_url, config.bitcoin_network)
+            .await
+            .expect("failed to create bitcoin wallet");
+    let bitcoin_balance = bitcoin_wallet.balance().await?;
+    info!(
+        "Connection to Bitcoin wallet succeeded, balance: {}",
+        bitcoin_balance
+    );
+    let bitcoin_wallet = Arc::new(bitcoin_wallet);
+
+    let monero_wallet = monero::Wallet::new(monero_wallet_rpc_url);
+    let monero_balance = monero_wallet.get_balance().await?;
+    info!(
+        "Connection to Monero wallet succeeded, balance: {}",
+        monero_balance
+    );
+    let monero_wallet = Arc::new(monero_wallet);
+
+    Ok((bitcoin_wallet, monero_wallet))
+}
+
+async fn alice_swap(
+    swap_id: Uuid,
+    state: AliceState,
+    listen_addr: Multiaddr,
+    bitcoin_wallet: Arc<bitcoin::Wallet>,
+    monero_wallet: Arc<monero::Wallet>,
+    config: Config,
+    db: Database,
+) -> Result<AliceState> {
+    let alice_behaviour = alice::Behaviour::default();
+
+    let alice_peer_id = alice_behaviour.peer_id();
+    info!("Own Peer-ID: {}", alice_peer_id);
+
+    let alice_transport = build(alice_behaviour.identity())?;
+
+    let (mut event_loop, handle) =
+        alice::event_loop::EventLoop::new(alice_transport, alice_behaviour, listen_addr)?;
+
+    let swap = alice::swap::swap(
+        state,
+        handle,
+        bitcoin_wallet.clone(),
+        monero_wallet.clone(),
+        config,
+        swap_id,
+        db,
+    );
+
+    let _event_loop = tokio::spawn(async move { event_loop.run().await });
+    Ok(swap.await?)
+}
+
+async fn bob_swap(
+    swap_id: Uuid,
+    state: BobState,
+    bitcoin_wallet: Arc<bitcoin::Wallet>,
+    monero_wallet: Arc<monero::Wallet>,
+    db: Database,
+) -> Result<BobState> {
+    let bob_behaviour = bob::Behaviour::default();
+    let bob_transport = build(bob_behaviour.identity())?;
+
+    let (event_loop, handle) =
+        bob::event_loop::EventLoop::new(bob_transport, bob_behaviour).unwrap();
```

Comment:
```suggestion
        bob::event_loop::EventLoop::new(bob_transport, bob_behaviour)?;
```
The function you are in returns a result.